new-lamassu-admin/src/pages/Customers/CustomerProfile.js

```diff
@@ -0,0 +1,378 @@
+import { ReactComponent as TxInIcon } from 'src/styling/icons/direction/cash-in.svg'
+import { ReactComponent as TxOutIcon } from 'src/styling/icons/direction/cash-out.svg'
+import { ReactComponent as BlockReversedIcon } from 'src/styling/icons/button/block/white.svg'
+import { ReactComponent as BlockIcon } from 'src/styling/icons/button/block/zodiac.svg'
+import { ReactComponent as AuthorizeReversedIcon } from 'src/styling/icons/button/authorize/white.svg'
+import { ReactComponent as AuthorizeIcon } from 'src/styling/icons/button/authorize/zodiac.svg'
+import { ReactComponent as CrossedCameraIcon } from 'src/styling/icons/ID/photo/crossed-camera.svg'
+import {
+  PropertyCard,
+  OVERRIDE_AUTHORIZED,
+  OVERRIDE_REJECTED
+} from 'src/components/propertyCard'
+
+import TransactionsList from './TransactionsList'
+import { mainStyles } from './Customers.styles'
+
+const useStyles = makeStyles(mainStyles)
+
+const GET_CUSTOMER = gql`
+  query customer($customerId: ID!) {
+    customer(customerId: $customerId) {
+      id
+      name
+      authorizedOverride
+      frontCameraPath
+      phone
+      smsOverride
+      idCardData
+      idCardDataOverride
+      idCardDataExpiration
+      idCardPhotoPath
+      idCardPhotoOverride
+      totalTxs
+      totalSpent
+      lastActive
+      lastTxFiat
+      lastTxFiatCode
+      lastTxClass
+    }
+  }
+`
+
+const GET_TRANSACTIONS = gql`
+  query customerTransactions($customerId: ID!) {
+    customerTransactions(customerId: $customerId) {
+      txClass
+      id
+      fiat
+      fiatCode
+      cryptoAtoms
+      cryptoCode
+      created
+      errorMessage: error
+      error: errorCode
+    }
+  }
+`
+
+const SET_CUSTOMER = gql`
+  mutation setCustomer($customerId: ID!, $customerInput: CustomerInput) {
+    setCustomer(customerId: $customerId, customerInput: $customerInput) {
+      id
+      name
+      authorizedOverride
+      frontCameraPath
+      phone
+      smsOverride
+      idCardData
+      idCardDataOverride
+      idCardDataExpiration
+      idCardPhotoPath
+      idCardPhotoOverride
+      totalTxs
+      totalSpent
+      lastActive
+      lastTxFiat
+      lastTxFiatCode
+      lastTxClass
+    }
+  }
+`
+
+const IMAGES_URI =
+  process.env.NODE_ENV === 'development' ? 'https://localhost:8070' : ''
+
+const CustomerDetails = memo(({ customer }) => {
+  const classes = useStyles()
+
+  const elements = [
+    {
+      header: 'Transactions',
+      size: 127,
+      value: `${Number.parseInt(customer.totalTxs)}`
+    },
+    {
+      header: 'Transaction volume',
+      size: 167,
+      value: `${Number.parseFloat(customer.totalSpent)} 
+        ${customer.lastTxFiatCode}`
+    },
+    {
+      header: 'Last active',
+      size: 142,
+      value: moment.utc(customer.lastActive).format('YYYY-MM-D')
+    },
+    {
+      header: 'Last transaction',
+      size: 198,
+      value: (
+        <>
+          {`${Number.parseFloat(customer.lastTxFiat)} 
```

Comment:
I think this is the stuff that can be causing all of the NaNs. Lets make a method to cleanly handle invalid numbers and use it instead of the parses under Number

Reply:
I think that all of the NaNs are caused by null values in the parse methods... should I just test for a null value prior to parsing it then?

Reply:
You should, what I'm saying is to create a method for this and use it everywhere instead of doing the test everywhere.